app/controllers/solidus_stripe/intents_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module SolidusStripe
+  class IntentsController < Spree::BaseController
+    include Spree::Core::ControllerHelpers::Order
+
+    def confirm
+      begin
+        if params[:stripe_payment_method_id].present?
+          intent = stripe.create_intent(
+            (current_order.total * 100).to_i,
+            params[:stripe_payment_method_id],
+            currency: current_order.currency,
+            confirmation_method: 'manual',
+            confirm: true,
+            setup_future_usage: 'on_session',
+            metadata: { order_id: current_order.id }
+          )
+        elsif params[:stripe_payment_intent_id].present?
+          intent = stripe.confirm_intent(params[:stripe_payment_intent_id], nil)
+        end
+      rescue Stripe::CardError => e
+        render json: { error: e.message }
```

Comment:
We should add the correct status code to this render, by default it's `200` even if they are errors. Something like:

```ruby
 render json: { error: e.message }, status: 500
```

Reply:
Still missing here?